client/ayon_core/plugins/load/export_otio.py

```diff
@@ -0,0 +1,280 @@
+
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.addWidget(input_widget)
+        layout.addWidget(export_widget)
+
+        self.button_export.clicked.connect(self.export)
+        self.button_output_path.clicked.connect(self.set_output_path)
+
+    def toggle_all(self):
+        representation_name = self.sender().text()
+        for item in self._representation_widgets[representation_name]:
+            item["widget"].setChecked(True)
+
+    def set_output_path(self):
+        file_dialog = QtWidgets.QFileDialog()
+        file_dialog.setNameFilter("OTIO Files (*.otio)")
+        file_dialog.setFileMode(QtWidgets.QFileDialog.ExistingFile)
+        if file_dialog.exec_():
+            selected_files = file_dialog.selectedFiles()
+            output_path = selected_files[0] if selected_files else None
+            if output_path:
+                self.lineedit_output_path.setText(output_path)
+
+    def export(self):
+        output_path = self.lineedit_output_path.text()
+
+        # Validate output path is not empty.
+        if not output_path:
+            show_message_dialog(
+                "Missing output path",
+                (
+                    "Output path is empty. Please enter a path to export the "
+                    "OTIO file to."
+                ),
+                level="critical",
+                parent=self
+            )
+            return
+
+        # Validate output path ends with .otio.
+        if not output_path.endswith(".otio"):
+            show_message_dialog(
+                "Wrong extension.",
+                (
+                    "Output path needs to end with .otio."
+                ),
+                level="critical",
+                parent=self
+            )
+            return
+
+        representations = []
+        for name, items in self._representation_widgets.items():
+            for item in items:
+                if item["widget"].isChecked():
+                    representations.append(item["representation"])
+
+        anatomy = Anatomy(self._project_name)
+        clips_data = {}
+        for representation in representations:
+            version = self._version_by_representation_id[representation["_id"]]
+            name = "{}/{}".format(
+                self._version_path_by_id[version["_id"]],
+                representation["name"]
+            )
+            clips_data[name] = {
+                "path": get_representation_path_with_anatomy(
+                    representation, anatomy
+                ),
+                "frames": (
+                    version["data"]["frameEnd"] -
+                    version["data"]["frameStart"]
+                ),
+                "framerate": version["data"]["fps"]
+            }
+
+        self.export_otio(clips_data, output_path)
+
+        check_state = self.checkbox_inspect_otio_view.checkState()
+        if check_state == QtCore.Qt.CheckState.Checked:
+            run_detached_process(["otioview", output_path])
```

Comment:
Some sanity check for presence of `otioview` would be desired. In my case it had crashed with:

```
Traceback (most recent call last):
  File "C:\CODE\__PYPE\ayon-core\client\ayon_core\plugins\load\export_otio.py", line 241, in export
    run_detached_process(["otioview", output_path])
  File "C:\CODE\__PYPE\ayon-core\client\ayon_core\lib\execute.py", line 302, in run_detached_process
    process = subprocess.Popen(args, **kwargs)
  File "C:\Python\pyenv-win\versions\3.9.4\lib\subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Python\pyenv-win\versions\3.9.4\lib\subprocess.py", line 1420, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

But I am having `otioview` inside of my vendor folder at ayon-launcher. 
![image](https://github.com/ynput/ayon-core/assets/40640033/512e8d14-eb2c-4a10-8f1f-f163750bd6f5)

Perhaps we should direct the path as abs to launcher's `otioview`?

Reply:
I had tested it with the abs path to my otioview from above comment and it seems we might have some buggy dependency:

```
Traceback (most recent call last):
  File "C:\Python\pyenv-win\versions\3.9.4\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python\pyenv-win\versions\3.9.4\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\CODE\__PYPE\ayon-launcher\vendor\python\bin\otioview.exe\__main__.py", line 4, in <module>
ModuleNotFoundError: No module named 'opentimelineview'
```

Reply:
But @iLLiCiTiT will tell you more about **OpenTimelineIO** dependency redesign.